include/uprotocol-cpp/uri/validator/UriValidator.h

```diff
@@ -19,21 +19,92 @@
  * under the License.
  * 
  * SPDX-FileType: SOURCE
- * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
+ * SPDX-FileCopyrightText: 2024 General Motors GTO LLC
  * SPDX-License-Identifier: Apache-2.0
  */
 #ifndef URI_VALIDATOR_H_
 #define URI_VALIDATOR_H_
 
 #include <uprotocol-cpp/uri/serializer/LongUriSerializer.h>
 #include <uprotocol-cpp/uri/tools/Utils.h>
+#include "uprotocol/uri.pb.h"
+#include "uprotocol/ustatus.pb.h"
+#include <string.h>
 
+using namespace uprotocol::v1;
 namespace uprotocol::uri {
+/* UriValidator class to validate uri*/
+    class UriValidator {
+        public:
 
-bool valid_uri(const std::string& uri) {
-    auto uri_view = LongUriSerializer::deserialize(uri);
-    return !isEmpty(uri_view);
-}
+         /**
+          * isValidUri a {@link uri} to ensure that it has at least a name for the
+          * uEntity.
+          *
+          * @param uri {@link uri} to validate.
+          * @return Returns bool containing a true or a false
+          */
+
+          static bool isValidUri(const std::string& uri);
+
+         /**
+          * Validate a {@link UUri} to ensure that it has at least a name for the
+          * uEntity.
+          *
+          * @param uri {@link UUri} to validate.
+          * @return Returns Status containing a success or a failure with the error
```

Comment:
Go ahead and put the end of this sentence on the same line.  Lines should be less than 120 characters.  Several other places in this file.

Reply:
Done